src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs

```diff
@@ -449,5 +449,63 @@ public static class BuildIntegratedRestoreUtility
 
             return closure;
         }
+
+        /// <summary>
+        /// Find the list of child projects direct or indirect references of target project in
+        /// reverse dependency order like the least dependent package first.
+        /// </summary>
+        public static async Task GetChildProjectsInClosure(BuildIntegratedNuGetProject target,
```

Comment:
You may want to test the perf on this, the noop restore was VS had code very similar to this but it was locking up the UI from all the DTE calls to get project references. It was fixed by caching the child projects everywhere possible.